lib/origen_arm_debug/sw_dp_controller.rb

```diff
@@ -39,6 +39,13 @@ def read_register(reg, options = {})
 
         select_ap_reg(reg)
         dut.swd.read_ap(address: reg.address)
+
+        # Add any extra delay needed in between selecting the AP state, initiating and completing a dummy read, and
+        # starting the actual read.
+        if options[:apacc_wait_states]
```

Comment:
So upstream for here something has read the apacc_wait_states attribute and put it into the options?

Reply:
Yes. It can be set directly after the block instantiation, or passed in along with the mem_ap's options. In our app, it looks like:

~~~ruby
      mem_aps = {
        ahbap:  { base_address: 0x00000000, apmem_access_wait: 20, csw_reset: 0x23000040, apacc_wait_states: 4 }
      }

sub_block :arm_debug, class_name: 'OrigenARMDebug::DAP', base_address: 0, 
                            mem_aps:    mem_aps,
                            jtag_aps:   jtag_ap,
                            aps:        aps
~~~

Reply:
OK, thanks, LGTM then